src/hm/hmRadio.h

```diff
@@ -332,6 +335,10 @@ class HmRadio {
             mTxBuf[len] = ah::crc8(mTxBuf, len);
             len++;
 
+            // set TX and RX channels
+            mTxChIdx = (mTxChIdx + 1) % RF_CHANNELS;
+            mRxChIdx = (mTxChIdx + 2) % RF_CHANNELS;
+
```

Comment:
hier werden jetzt auf einmal RX und TX channel gewechselt, ist das so beabsichtigt?

Reply:
bei TX ist es egal, der soll nur bei jeder Benutzung gewechselt werden,
bei RX ist es wichtig, dass er TX + 2 (modulo RF_CHANNELS) ist, sonst wird der erste Frame selten erwischt. Hab' alle Möglichkeiten TX bis TX + 4 ausprobiert, TX + 2 liefert die besten Ergebnisse. Auch bessere als bisher, wo RX immer auf Kanal 03 anfängt. Wollte möglichst wenig Zeit **nach** dem Senden und **vor** dem Empfangen verlieren, daher die Anweisungen **vor** dem Senden. In der loop wird RX dann wie gewohnt nach Zeitschema auf den nächsten Kanal gewechselt. Da gibt es (bis auf stop / start listening) nur den Unterschied in der Zeitmessung. 4088us liefert bessere Ergebnisse als 4ms = 4000us.
Hast du dir schon die Plauderei mit rejoe in #776 angesehen?
In Kurzfassung: habe mir die Programmierung von Nordic Gazell angesehen. Danach wechselt der Empfänger (bei uns Ahoy) stur nach vorgegebener Zeit (timeslot) die Kanäle (sind als Default bei Gazell genau unsere!). Der Sender (bei uns der Inverter) probiert für die erste Message alle Kanäle aus. Sobald er ein ACK erhält, weiß er, auf welchem Kanal der Empfänger gerade ist und rechnet sich aus, auf welchem Kanal die nächste Message gesendet werden muss (laut Gazell: synchronisiert sich). Nach gewisser Zeit (Nordic empfiehlt < 1 Sekunde) fällt er aber wieder in den asynchronen Zustand zurück.
D.h. um TX müssen wir uns keine Gedanken machen, da wir nur eine Message senden. RX läuft schon immer nach dem Gazell Muster (unbewusst?), nur der RX Startkanal und der timeslot passen noch nicht ganz. Habe einige timeslots ausprobiert. Auch ein paar zusammen mit rejoe. Letztlich sind wir bei 4088us gelandet (s. #776).
Das scheint im Moment das beste zu sein. Wenn ich viel Zeit habe, will ich versuchen, einen zweiten NRF an den ESP anzuschließen um auf dem mit RX+1 zu lauschen. Damit könnte der "wahre" timeslot vielleicht genauer bestimmt werden. Im Moment kann ich aber nicht absehen, wann ich dazu komme.

Reply:
danke für die Zusammenfassung. Ich kann leider aktuell nicht so viel Zeit in Ahoy investieren und dadurch nicht alles lesen. Es wird leider immer mehr Content erzeugt (Issues, Kommentare), sodass man einen ganzen Tag mit lesen und verstehen verbringen könnte 😃. Finde toll, dass hier so stark zusammengearbeitet wird.
Sehr interessant was ihr da rausgefunden habt. Ich werde es einfach testen und wahrscheinlich die Tage dann mergen. Ich konnte hier nicht rauslesen, dass deine Erkenntnisse hier auch reingeflossen sind.